pkg/api/api.go

```diff
@@ -74,6 +74,8 @@ type Service interface {
 	Events(ctx context.Context, projectName string, options EventsOptions) error
 	// Port executes the equivalent to a `compose port`
 	Port(ctx context.Context, projectName string, service string, port uint16, options PortOptions) (string, int, error)
+	// Publish executes the equivalent to a `compose publish`
+	Publish(ctx context.Context, project *types.Project, repository string) error
```

Comment:
Minor: we often end up refactoring the API methods to add an options object later...maybe worth having `PublishOptions` with a `Repository string` field upfront

Reply:
e.g. I'm guessing this will need `Quiet bool` at some point

Reply:
Repository being mandatory parameter should not be part of the `Options` struct imho